cognite/pygen/_core/models/data_classes.py

```diff
@@ -325,15 +325,6 @@ def pydantic_field(self) -> Literal["Field", "pydantic.Field"]:
         """
         if any(
             name == "Field" for name in [self.read_name, self.write_name, self.read_list_name, self.write_list_name]
-        ) or any(
-            name == "Field"
-            for dependency in self.dependencies
-            for name in [
-                dependency.read_name,
-                dependency.write_name,
-                dependency.read_list_name,
-                dependency.write_list_name,
-            ]
         ):
```

Comment:
This was the issue. When the `.update_fields()` get called this property changed through the iterations.